src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/ForwardCancellationTokenToInvocations.Analyzer.cs

```diff
@@ -0,0 +1,338 @@
+        {
+            if (methodDeclaration.Parameters.Count(x => x.Type.Equals(cancellationTokenType)) == 1 &&
+                methodDeclaration.Parameters[^1] is IParameterSymbol lastParameter &&
+                lastParameter.Type.Equals(cancellationTokenType)) // Covers the case when using an alias for ct
+            {
+                cancellationTokenParameterName = lastParameter.Name;
+                return true;
+            }
+
+            cancellationTokenParameterName = null;
+            return false;
+        }
+
+        // Checks if the invocation has an optional ct argument at the end or a params ct array at the end.
+        private static bool InvocationMethodTakesAToken(
+            IMethodSymbol method,
+            ImmutableArray<IArgumentOperation> arguments,
+            INamedTypeSymbol cancellationTokenType)
+        {
+            return
+                !method.Parameters.IsEmpty &&
+                method.Parameters[^1] is IParameterSymbol lastParameter &&
+                (InvocationIgnoresOptionalCancellationToken(lastParameter, arguments, cancellationTokenType) ||
+                InvocationIsUsingParamsCancellationToken(lastParameter, arguments, cancellationTokenType));
+        }
+
+        // Checks if the arguments enumerable has any elements that satisfy the provided condition,
+        // starting the lookup with the last element since tokens tend to be added as the last argument.
+        private static bool AnyArgument(ImmutableArray<IArgumentOperation> arguments, Func<IArgumentOperation, bool> predicate)
+        {
+            for (int i = arguments.Length - 1; i >= 0; i--)
+            {
+                if (predicate(arguments[i]))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        // Check if the currently used overload is the one that takes the ct, but is utilizing the default value offered in the method signature.
+        // We want to offer a diagnostic for this case, so the user explicitly passes the ancestor's ct.
+        private static bool InvocationIgnoresOptionalCancellationToken(
+            IParameterSymbol lastParameter,
+            ImmutableArray<IArgumentOperation> arguments,
+            INamedTypeSymbol cancellationTokenType)
+        {
+            if (lastParameter.Type.Equals(cancellationTokenType) &&
+                lastParameter.IsOptional) // Has a default value being used
+            {
+                // Find out if the ct argument is using the default value
+                // Need to check among all arguments in case the user is passing them named and unordered (despite the ct being defined as the last parameter)
+                return AnyArgument(
+                    arguments,
+                    a => a.Parameter.Equals(cancellationTokenType) && a.ArgumentKind == ArgumentKind.DefaultValue);
```

Comment:
I am not sure how this test is succeeding - you are comparing a parameter to a type in `a.Parameter.Equals(cancellationTokenType)`

Should this be:
```suggestion
                    a => a.Parameter.Equals(lastParameter) && a.ArgumentKind == ArgumentKind.DefaultValue);
```

Does this code path need a unit test?

Reply:
I applied your previous suggestion: https://github.com/dotnet/roslyn-analyzers/pull/3641#discussion_r436920118

Reply:
Fixed.